Controller/ResourceController.php

```diff
@@ -39,11 +46,103 @@ public function __construct(
         $this->registry = $registry;
     }
 
-    public function resourceAction($repositoryName, $path)
+    public function getResourceAction($repositoryName, $path)
     {
         $repository = $this->registry->get($repositoryName);
         $resource = $repository->get('/'.$path);
 
+        return $this->createResponse($resource);
+    }
+
+    /**
+     * There are two different changes that can currently be done on a cmf resource:
+     *
+     * - move
+     * - rename
+     *
+     * changing payload properties isn't supported yet.
+     *
+     * @param string $repositoryName
+     * @param string $path
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function patchResourceAction($repositoryName, $path, Request $request)
+    {
+        $repository = $this->registry->get($repositoryName);
+        $this->failOnNotEditable($repository, $repositoryName);
+
+        $resourceName = $request->get('node_name');
+
+        $targetPath = null;
+        if ($path !== $path) {
```

Comment:
To be honest, I don't understand much of the code in this controller. Isn't this what you want:
```php
function patchResourceAction($repositoryName, $path, Request $request)
{
    $repository = $this->registry->get($repositoryName);
    $this->failOnNotEditable($repository, $repositoryName);

    $targetPath = $request->get('node_name');

    try {
        $repository->move($path, $targetPath);
    } catch (\Exception $e) {
        return $this->createBadRequestResponse($e->getMessage()); // why not rely on Symfony's great error handler here?
    }

    $resource = $repository->get($targetPath);

    return $this->createResponse($resource); // not sure whether we should really return the resource here, I think we should sent a 200 response without body
}
```